### PR TITLE
LPS-177007 Fix has/has-not and Entity input selectors

### PR DESCRIPTION
**Pull request by Stephy6**
Motivation
=======
This PR contains is a continuation of [this](https://github.com/liferay-tango/liferay-portal/pull/3269) one. It adds the capability change the value for has/has-not input.
[Link to the task](https://issues.liferay.com/browse/LPS-177007)
[Link to the story](https://issues.liferay.com/browse/LPS-173020)

Proposed Solution
========
The solution consist in handle the case when the has/has-not input changed and modify the value sent to the backend which has to be a boolean and not a string with the `has` or `has-not` values.

Steps to Verify:
----------------
1. Connect to AC to DXP.
2. Go to AC -> Segments -> Dynamic Segment -> Events -> Downloaded Documents and Media and create a segment.
3. Go to Applications Menu -> Control Menu -> Job Scheduler -> Scheduled Jobs -> (page 2) -> Click on Run Now for the CheckIndividualSegmentsMessageListener to sync up the segments between AC and DXP 
4.  Go to ProductMenuPanel -> People -> Segments
5. Edit the segment created in AC by clicking on it and clicking on the toggle button
6. Click on the select button that opens a modal with the DXP documents. 
7. Change the at has/has-not inputs.

### CODE REVIEW

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/SelectEntityInput.es.js [hunk 1]
```diff
@@ -106,6 +106,7 @@ class SelectEntityInput extends React.Component {
 							'criterion-input--error':
 								!value && renderEmptyValueErrors,
 						})}
+						data-testid="entity-select-displayed-input"
```
Comment (claraizquierdo) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/SelectEntityInput.es.js:109]: I think that this testid is not being used.

**Review by claraizquierdo — COMMENTED** [review]
Hi @Stephy6 ,

everything is working perfectly :D I just added one comment because I think that one testid is not needed.

**Comment by Stephy6** [comment]
ci:test:sf

**Comment by Stephy6** [comment]
ci:test:relevant

**Comment by Stephy6** [comment]
ci:forward

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/SelectEntityInput.es.js (hunk 1 above)
Comment (Stephy6) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/SelectEntityInput.es.js]: Done! thank you

**Pull request closed without merging**